lib/puppet/provider/package/version/range.rb

```diff
@@ -0,0 +1,102 @@
+require 'puppet/provider/package/version'
+class Puppet::Provider::Package::Version::Range
```

Comment:
this can be written as
```
module Puppet::Provider::Package::Version
  class Range
```

then `'puppet/provider/package/version'` won't be needed

Reply:
Is this the same that we discussed and it brakes rake spec task?

Reply:
Sadly, yes. Seems like having those under provider is not the happiest location and something gets messed when puppet is loaded in spec